### ISSUE TITLE
Feature request: Inheriting fields in subclasses

### ISSUE THREAD

**Issue by ElectronicRU** [bckohan/enum-properties#74]
Tried to have a common class from which several EnumProperties classes inherit. It doesn't work:

```python
class Symbol(StrEnumProperties):
    min_args: int
    max_args: int

class FunctionSymbol(Symbol):
    ADD = '+', 2, 2
```

The legacy way, with special base classes, doesn't appear to work either.
Easy enough to work around, but fairly unfortunate. Is this a reasonable feature to add?

**Comment by bckohan** [comment]
Hmm, interesting. I wonder if something like this would work now:

```
class Symbol:
    min_args: int
    max_args: int

class FunctionSymbol(Symbol, StrEnumProperties):
    ADD = '+', 2, 2
```

**Comment by ElectronicRU** [comment]
I don't believe so, looking at the code - parent classes' `__annotations__` aren't checked anywhere

**Comment by bckohan** [comment]
I'm worried about the unintended consequences of this. Presumably there are bunch of EnumProperties in the wild that inherit from other classes. If those classes have or add class member type hints it'll break the enums.

Is this use case mostly convenience? If there is a compelling need for it, I could see supporting something like:

```python

class Symbol:
    min_args: int
    max_args: int

class FunctionSymbol(StrEnumProperties, properties=Symbol):
    ADD = '+', 2, 2

```

**Comment by ElectronicRU** [comment]
I was thinking about that, `django-enum` `choices.TextChoices` would be an example if inheritance is uncapped.

What about limiting it to cases where no annotations or legacy bases are provided for the leaf class, then search for base classes? I cannot imagine a usecase for that.

Re your example, it kind of breaks static typing.

**Comment by bckohan** [comment]
The potential for hard to understand unintended consequences here is quite high. This is why ``enum.Enum`` does not support inheritance that adds to the enum values. So if properties added from base classes are supported I think it needs to be explicit. For example, to solve the static typing issue you mentioned you could do something like this:

```python

class Biased:
    bias: float    

class Symbol:
    min_args: int
    max_args: int

class FunctionSymbol(Symbol, Biased, StrEnumProperties, properties=[Symbol, Biased]):
    ADD = '+', 2, 2, 1.1

```

Still seems like its a solution in search of a problem to me though.
